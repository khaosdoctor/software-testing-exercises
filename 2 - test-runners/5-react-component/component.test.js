import renderer from 'react-test-renderer'
import App from './component'

describe('Errors', () => {
  it('does not render if the name is not a string', () => {
    expect(() => renderer.create(<App name={1} />)).toThrow(new Error('Name is not a string'))
  })
})

describe('Happy path', () => {
  it('correctly renders a name', () => {
    const component = renderer.create(<App name="Lucas" />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('correctly renders default name', () => {
    const component = renderer.create(<App />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
