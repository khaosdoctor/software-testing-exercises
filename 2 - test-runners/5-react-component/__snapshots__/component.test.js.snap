// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Happy path correctly renders a name 1`] = `
<div>
  <div>
    <p>
      Hello 
      Lucas
      !
    </p>
  </div>
</div>
`;

exports[`Happy path correctly renders default name 1`] = `
<div>
  <div>
    <p>
      Hello 
      World
      !
    </p>
  </div>
</div>
`;
